Azure/azure-service-operator
# Create KeyVault Key operators for creating encryption keys in keyvault

closes #642 
closes #643 

Creates a new type for KeyVaultKey 
Creates an ARMClient implementation for KeyVaultKey

To test:
- Create a resource group
- deploy a keyvault to the new rg with get/create/delete permissions for Keys
- use config/sample/ file for keyvaultkey to create a key in new keyvault
    - verify key in portal
- delete key
    - verify in portal

Also consider edge cases like what happens when the keyvault does not exist or the resourcegroup does not exist

What happens when the keyvault has incorrect access policies to allow the operator to create a key?
